eng/Versions.props

```diff
@@ -20,8 +20,8 @@
     <ReportGeneratorVersion>4.3.6</ReportGeneratorVersion>
 
     <!-- Roslyn -->
-    <MicrosoftCodeAnalysisVersion>3.0.0</MicrosoftCodeAnalysisVersion>
-    <MicrosoftNetCompilersVersion>3.4.0-beta2-final</MicrosoftNetCompilersVersion>
+    <MicrosoftCodeAnalysisVersion>3.5.0-beta2-19619-02</MicrosoftCodeAnalysisVersion>
```

Comment:
I don't think we want to move all of the repo to MS.CA 3.5. Can we just move PublicApiAnalyzers to the newer version: https://github.com/dotnet/roslyn-analyzers/blob/master/src/PublicApiAnalyzers/Core/Analyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.csproj#L10?

Note this is required to allow us to release analyzer packages from the repo that work on VS2019 16+ from this branch.

Reply:
Also tagging @sharwell since he has made conscious efforts to keep the PublicApiAnalyzers on MS.CA 1.2.1.

Reply:
Do you need the new version just for the tests? Seems like the analyzer does not use any APIs? Also should nullable annotations be opt-in?

Reply:
I'll look at narrowing down the dependency on new package. It should indeed only be used for testing.

I think that nullability annotations should always be part of public API files because changing annotations can cause compat breaks. But I'm open to discussing this. 
What do you think?

Reply:
> Also tagging @sharwell since he has made conscious efforts to keep the PublicApiAnalyzers on MS.CA 1.2.1.

We should only update the dependency if both of the following hold:

1. We need to use an API from the newer version
2. We cannot use the newer API via reflection

Reply:
> Also should nullable annotations in public API files be opt-in?

I don't see any need to have this be an option. The API files won't change if NRT is disabled, and if NRT is enabled then it only makes sense to update the API files too.

Reply:
> if NRT is enabled then it only makes sense to update the API files too.

This will mean an enforced breaking change on consumers of this analyzer. I am fine with going that route as long as it is a conscious decision.

Reply:
Yes, this addition to shipped API files would break folks who start using nullability and then upgrade to newer analyzer. If we're ever going to do this, it should be earlier rather than later (most projects probably not using nullability yet).

Reply:
@mavasani @tmat @sharwell Jared convinced me that making this change conditional would be preferable (as Manish suggested earlier), based on our own experience annotating compiler and IDE code.

I'm thinking that we can just use a marker `#nullable enable` in shipped API files to configure that. Sounds okay?

Reply:
I just want to record a good idea from @jasonmalinowski: we could later on add a `#nullable enable force` (or maybe `#nullable enable strict`) option, which would give an error if any oblivious public API is encountered.